Jenkinsfile

```diff
@@ -0,0 +1,142 @@
+#!groovy
+import groovy.json.JsonOutput
+
+repoName = 'realm-js' // This is a global variable
+
+def getSourceArchive() {
+  checkout scm
+  sh 'git clean -ffdx -e .????????'
+  sshagent(['realm-ci-ssh']) {
+    sh 'git submodule update --init --recursive'
+  }
+}
+
+def readGitTag() {
+  sh "git describe --exact-match --tags HEAD | tail -n 1 > tag.txt 2>&1 || true"
+  def tag = readFile('tag.txt').trim()
+  return tag
+}
+
+def readGitSha() {
+  sh "git rev-parse HEAD | cut -b1-8 > sha.txt"
+  def sha = readFile('sha.txt').readLines().last().trim()
+  return sha
+}
+
+def getVersion(){
+  def dependencies = readProperties file: 'dependencies.list'
+  def gitTag = readGitTag()
+  def gitSha = readGitSha()
+  if (gitTag == "") {
+    return "${dependencies.VERSION}-g${gitSha}"
+  }
+  else {
+    return dependencies.VERSION
+  }
+}
+
+def setBuildName(newBuildName) {
+  currentBuild.displayName = "${currentBuild.displayName} - ${newBuildName}"
+}
+
+def gitTag
+def gitSha
+def dependencies
+def version
+
+stage('check') {
+  node('docker') {
+    getSourceArchive()
+
+    dependencies = readProperties file: 'dependencies.list'
+
+    gitTag = readGitTag()
+    gitSha = readGitSha()
+    version = getVersion()
+    echo "tag: ${gitTag}"
+    if (gitTag == "") {
+      echo "No tag given for this build"
+      setBuildName("${gitSha}")
+    } else {
+      if (gitTag != "v${dependencies.VERSION}") {
+        echo "Git tag '${gitTag}' does not match v${dependencies.VERSION}"
+      } else {
+        echo "Building release: '${gitTag}'"
+        setBuildName("Tag ${gitTag}")
+      }
+    }
+    echo "version: ${version}"
+  }
+}
+
+def reportStatus(target, state, message) {
```

Comment:
Nice. Though i guess that if we still see issues with the GitHub api rate limit, we might want to disable it